### PR TITLE
Processing 'Additionals' section of query response

### PR DESCRIPTION
**Pull request by davidcreager**
I found that the IP address of a device was not returned, after investigation I found that the 'A' Record was in the Additionals section and not the Answers section. Apologies, I made style changes that helped me to debug.

I implemented a function to process each record  to avoid duplicating the code as Additionals and Answers are the same format (I believe)

Hope this helps

### CODE REVIEW

File: mdns.lua [hunk 1]
```diff
@@ -103,66 +89,43 @@ local function mdns_parse(service, data, answers)
...
-            end
+            if (rdlength ~= 4) then return nil, 'bad RDLENGTH with A record' end
             answers.a[name] = string.format('%d.%d.%d.%d', data:byte(rdoffset + 0), data:byte(rdoffset + 1), data:byte(rdoffset + 2), data:byte(rdoffset + 3))
         end
-
         -- PTR record (pointer)
         if (type == 12) then
             local target = parse_name(data, rdoffset)
-            table.insert(answers.ptr, target)
+            table.insert(answers.ptr, target)
```
Comment (JaredSartin) [inline mdns.lua]: There seems to be extra trailing space here

File: mdns.lua [hunk 2]
```diff
@@ -41,10 +41,7 @@
 ]]--
 
 local mdns = {}
-
 local socket = require('socket')
-
-
```
Comment (JaredSartin) [inline mdns.lua]: Ideally there are line breaks between non-associated blocks to help readability. However, I wasn't the original author, so I am less picky

Were the inline if/returns and whitespace changes done with a code formatter or linter?

Comment (davidcreager) [reply mdns.lua]: No......want me to undo?

Comment (JaredSartin) [reply mdns.lua]: I would appreciate it. Happy to commit the specific improvements you made, but a code-style refactor + code change is harder to validate and approve.

Comment (davidcreager) [reply mdns.lua]: Argh...sorry I am crap with git.....I'm sure I could have made it easier, but anyway I have removed the line spaces etc
Hope this is better

**Review by JaredSartin — APPROVED** [review]

**Pull request merged**